src/detect-engine-payload.c

```diff
@@ -229,21 +220,17 @@ static int StreamContentInspectFunc(
         void *cb_data, const uint8_t *data, const uint32_t data_len, const uint64_t _offset)
 {
     SCEnter();
-    int r = 0;
     struct StreamContentInspectData *smd = cb_data;
 #ifdef DEBUG
     smd->det_ctx->stream_persig_cnt++;
     smd->det_ctx->stream_persig_size += data_len;
 #endif
-    smd->det_ctx->buffer_offset = 0;
-    smd->det_ctx->discontinue_matching = 0;
-    smd->det_ctx->inspection_recursion_counter = 0;
 
-    r = DetectEngineContentInspection(smd->de_ctx, smd->det_ctx,
-            smd->s, smd->s->sm_arrays[DETECT_SM_LIST_PMATCH],
-            NULL, smd->f, (uint8_t *)data, data_len, 0, 0, //TODO
+    const bool match = DetectEngineContentInspection(smd->de_ctx, smd->det_ctx, smd->s,
+            smd->s->sm_arrays[DETECT_SM_LIST_PMATCH], NULL, smd->f, (uint8_t *)data, data_len, 0,
+            0, // TODO
```

Comment:
What is this todo related to? Can it be addressed or removed?

Reply:
I'm not entirely sure, but I think its about setting `DETECT_CI_FLAGS_*`